src/celeritas/user/SDParams.hh

```diff
@@ -34,17 +35,14 @@ class SDParams final : public ParamsDataInterface<SDParamsData>
     //!@}
 
   public:
-    //! Default constructor: no detectors
-    SDParams() = default;
-
     //! Construct from canonical volume IDs
-    SDParams(GeoParamsInterface const& geo, VecVolId&& volume_ids);
+    SDParams(CoreGeoParams const& geo, inp::Detectors detectors);
 
     //! Whether any detectors are present
     bool empty() const { return !static_cast<bool>(mirror_); }
 
     //! Number of detectors
-    DetectorId::size_type size() const { return volume_ids_.size(); }
+    DetectorId::size_type size() const { return detectors_.detectors.size(); }
```

Comment:
`detectors.detectors` sounds silly, even though I know what's going on. Just wondering if there is a better naming for the vector in `inp::Detectors`...